From: Jan Christoph Nordholz <hesso@pool.math.tu-berlin.de>
Date: Wed, 5 Aug 2026 16:06:32 +0200
Subject: _session_creation_docs

Bug-Debian: https://bugs.debian.org/206572
Forwarded: no
Last-Update: 2026-05-20

Description: Document session-creation timestamp behaviour for -ls and -RR
---
 doc/screen.1       | 6 ++++--
 doc/screen.texinfo | 8 +++++---
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/doc/screen.1 b/doc/screen.1
index f8f51cb..c73c904 100644
--- a/doc/screen.1
+++ b/doc/screen.1
@@ -244,7 +244,7 @@ does not start
 .IR screen ,
 but prints a list of
 .I pid.tty.host
-strings identifying your
+strings and creation timestamps identifying your
 .I screen
 sessions.
 Sessions marked `detached' can be resumed with \*Qscreen \-r\*U. Those marked
@@ -358,7 +358,7 @@ when only one
 .I screen
 is detached. Otherwise lists available sessions.
 .B \-RR
-attempts to resume the first detached
+attempts to resume the youngest (in terms of creation time) detached
 .I screen
 session it finds.  If successful, all other command-line options are ignored.
 If no detached session exists, starts a new session using the specified
@@ -368,6 +368,8 @@ had not been specified. The option is set by default if
 .I screen
 is run as a login-shell (actually screen uses \*Q\-xRR\*U in that case).
 For combinations with the \fB\-d\fP/\fB\-D\fP option see there.
+.B Note:
+Time-based session selection is a Debian addition.
 .TP 5
 .BI "\-s " program 
 sets the default shell to the program specified, instead of the value
diff --git a/doc/screen.texinfo b/doc/screen.texinfo
index c142100..46d7426 100644
--- a/doc/screen.texinfo
+++ b/doc/screen.texinfo
@@ -399,7 +399,8 @@ is equivalent to the @code{deflogin} command (@pxref{Login}).
 @itemx -list [@var{match}]
 Do not start @code{screen}, but instead print a list of session
 identification strings (usually of the form @var{pid.tty.host};
-@pxref{Session Name}).  Sessions marked @samp{detached} can be resumed
+@pxref{Session Name}) and the corresponding creation timestamps.
+Sessions marked @samp{detached} can be resumed
 with @code{screen -r}.  Those marked @samp{attached} are running and
 have a controlling terminal.  If the session runs in multiuser mode,
 it is marked @samp{multi}.  Sessions marked as @samp{unreachable} either
@@ -499,14 +500,15 @@ resumes screen only when it's unambiguous which one to attach, usually
 when only one @code{screen} is detached. Otherwise lists available sessions.
 
 @item -RR
-Resume the first appropriate detached @code{screen} session.  If
-successful, all other command-line options are ignored.  If no detached
+Resume the most-recently created appropriate detached @code{screen} session.
+If successful, all other command-line options are ignored.  If no detached
 session exists, start a new session using the specified options, just as
 if @samp{-R} had not been specified.  This option is set by default if
 screen is run as a login-shell (actually screen uses @samp{-xRR} in
 that case).
 For combinations with the 
 @samp{-D}/@samp{-d} option see there.
+@samp{Note:} Time-based session selection is a Debian addition.
 
 @item -s @var{program}
 Set the default shell to be @var{program}.  By default, @code{screen}
